### DISCUSSION TITLE
스터디 정보를 가져오는 api를 하나만 사용하고 query parameter에 다이나믹하게 값을 넣는것

### DISCUSSION THREAD

**Discussion by airman5573** [woowacourse-teams/2022-moamoa#26] [토론]
(1) moamoa.com/api/studies?page={number}&size={number} : 데이터를 가져오는 용도  

(2) moamoa.com/api/studies/search?page={number}&size={number}&title={string}&date={string}&... : 데이터를 가져오는 용도

이렇게 둘을 나누는게 좋을지, 아니면  

moamoa.com/api/studies/search?page={number}&size={number} : 로 (1)번의 경우를 커버하고, 필터링 요소에 따라서 (2) 번으로도 사용하는건 어떨지 궁금합니다 :D 

태태 의견(@nan-noo) : 기능이 분리되는게 좋을것 같다. (1) / (2) 따로따로 api를 만들자.

병민 의견 : 위와 동일.

**Comment by verus-j** [comment]
기능이 분리되는 것이 좋을 것 같다
하나의 api로 구현하게 되면 여러가지 책임을 가지게 되서 변경할 이유가 많이 생길 가능성이 생긴다.
추후, 유지보수를 위해서 api를 분리하여 각 api별로 책임을 가지도록 하는 것이 좋을 것 같다.

**Comment by jaejae-yoo** [comment]
api를 분리하는 것이 좋을 것 같습니다.
여러 기능을 담당하는 범용적인 api를 생성하면, 각자 기능을 나누어서 개발을 진행했을 때, 코드 변경에 대한 영향을 많이 받을 수 있다고 생각합니다.
